src/AsyncStateMachine/StateMachine.cs

```diff
@@ -82,30 +78,40 @@ public StateMachine(TState? initialState = null)
         #region IStateMachine implementation
 
         /// <inheritdoc/>
-        public async Task InitializeAsync(TState initialState)
+        public async Task InitializeAsync(TState? state = null)
         {
+            if (state.HasValue && !_configuration.HasState(state.Value))
+                throw new ArgumentException("State not configured");
```

Comment:
In my opinion, it's more an `InvalidOperationException`, as the argument itself is not problematic here.